### ISSUE TITLE
The last character is dropped when a string reaches a certain length

### ISSUE THREAD

**Issue by yitam** [Microsoft/msphpsql#388]
This problem is reproducible in a Linux environment with the latest odbc driver 13.1.7 (no such issue with the previous odbc drivers). 

The Repro Script

```
$field_type = 'VARCHAR(386)';
$inValue1 = str_repeat( "A", 385);

$stmt = sqlsrv_query( $conn, "CREATE PROC [TestFullLenStringsOut] (@p1 " . $field_type . ", @p2 " . $field_type . " OUTPUT)
 AS
 BEGIN
   SELECT @p2 = CONVERT(" . $field_type . ", @p1 + 'X')
 END" );
if( $stmt === false ) {
    die( print_r( sqlsrv_errors(), true ));
}

$stmt = sqlsrv_query($conn, "{CALL [TestFullLenStringsOut] (?, ?)}", 
     array(
       array($inValue1, SQLSRV_PARAM_IN, SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_CHAR), SQLSRV_SQLTYPE_VARCHAR(386)),
       array(&$outValue1, SQLSRV_PARAM_INOUT, SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_CHAR), SQLSRV_SQLTYPE_VARCHAR(386))));
if( $stmt === false ) {
    die( print_r( sqlsrv_errors(), true ));
}
while( sqlsrv_next_result( $stmt )) {}
echo "strlen: ";
print_r( strlen( $outValue1 ));
echo "\nsubstr: ";
var_dump( substr( $outValue1, -2, 2 ));
var_dump( substr( $outValue1, -10, 10 ));
```

Output:

```
strlen: 386
substr: string(2) "A"
string(10) "AAAAAAAAA"
```

However, using the script above but reduce the length from 386 to 385, the problem goes away. The output is shown below. In fact, there's seemingly some issue with the length 386. This is reproducible with any string that is over 385 characters.

```
strlen: 385
substr: string(2) "AX"
string(10) "AAAAAAAAAX"
```

**Comment by yitam** [comment]
Filed an ODBC issue in [connect](https://connect.microsoft.com/SQLServer/feedback/EditFeedback.aspx?FeedbackID=3134175)

**Comment by yitam** [comment]
The issue is fixed in the latest msodbc 13.1.9
